libyul/AsmParser.cpp

```diff
@@ -158,43 +158,49 @@ Statement Parser::parseStatement()
 		}
 		while (currentToken() == Token::Comma);
 
-		expectToken(Token::Colon);
-		expectToken(Token::Assign);
+		expectToken(Token::AssemblyAssign);
+
+		assignment.value.reset(new Expression(parseExpression()));
+		assignment.location.end = locationOf(*assignment.value).end;
+		return Statement{std::move(assignment)};
+	}
+	case Token::AssemblyAssign:
+	{
+		if (elementary.type() != typeid(Identifier))
+			fatalParserError("variable name must precede \":\".");
```

Comment:
Shouldn't this state `must precede ":="` ?

Reply:
Also it seems there are not tests for this?

Reply:
> Shouldn't this state `must precede ":="` ?

I didn't think about that, I just used the original string as much as possible.

> Also it seems there are not tests for this?

That's true, I'll add some